include/revng/Pipeline/Location.h

```diff
@@ -0,0 +1,181 @@
+#pragma once
+
+//
+// This file is distributed under the MIT License. See LICENSE.md for details.
+//
+
+#include <array>
+#include <optional>
+#include <string>
+#include <string_view>
+#include <type_traits>
+
+#include "llvm/ADT/StringRef.h"
+#include "llvm/Support/YAMLTraits.h"
+
+#include "revng/ADT/STLExtras.h"
+#include "revng/Support/MetaAddress.h"
+#include "revng/Support/YAMLTraits.h"
+
+namespace pipeline {
+
+namespace concepts {
+
+template<typename Type>
+concept Rank = std::is_same_v<typename Type::RankTag, Type>;
+
+template<typename Type>
+concept RootRank = Rank<Type> && Type::Depth == 0;
+
+namespace detail {
+
+template<typename From, typename To>
+struct RechabilityHelper {
```

Comment:
Documentation: reachability of what?

Reply:
Done: https://github.com/revng/revng/pull/197/files#diff-5e12934676b4472008086bc5d11b23bee4f9c170fd7a3594ba9d94ad73509958R119